### PR TITLE
docs(GCS+gRPC): add quickstart with DP enabled

### PR DESCRIPTION
**Pull request by coryan**
This is going to be used in the service documentation.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11767)

### CODE REVIEW

File: google/cloud/storage/examples/storage_grpc_samples.cc [hunk 1]
```diff
@@ -98,7 +113,11 @@ void AutoRun(std::vector<std::string> const& argv) {
   std::cout << "Running GrpcReadWrite() example" << std::endl;
   GrpcReadWriteCommand({bucket_name});
 
-  std::cout << "Running GrpcReadWrite() example" << std::endl;
+  // The DP example requires running on a GCE instance with DP enabled.
+  std::cout << "Running GrpcReadWriteDP() example" << std::endl;
```
Comment (devbww) [inline google/cloud/storage/examples/storage_grpc_samples.cc:117]: s/GrpcReadWriteDP/GrpcClientWithDP/

File: google/cloud/storage/examples/storage_grpc_samples.cc [hunk 2]
```diff
@@ -98,7 +113,11 @@ void AutoRun(std::vector<std::string> const& argv) {
...
+  GrpcClientWithDPCommand({});
+
+  std::cout << "Running GrpcClientWithProjectCommand() example" << std::endl;
```
Comment (devbww) [inline google/cloud/storage/examples/storage_grpc_samples.cc:120]: s/GrpcClientWithProjectCommand/GrpcClientWithProject/

File: google/cloud/storage/examples/storage_grpc_samples.cc [hunk 3]
```diff
@@ -72,10 +70,27 @@ void GrpcClientWithProject(std::string project_id) {
...
+void GrpcReadWriteCommand(std::vector<std::string> argv) {
+  if (argv.size() != 1 || argv[0] == "--help") {
+    throw examples::Usage("grpc-read-write <bucket-name>");
+  }
+  GrpcReadWrite(argv[0]);
+}
+
+void GrpcClientWithDPCommand(std::vector<std::string> const& argv) {
+  if (argv.empty()) return GrpcClientWithDP();
+  throw examples::Usage("grpc-client-with-dp");
```
Comment (devbww) [inline google/cloud/storage/examples/storage_grpc_samples.cc:91]: Nit: Change the order to match the others so the exception is not the flow-through case?

File: google/cloud/storage/examples/storage_grpc_samples.cc [hunk 4]
```diff
@@ -13,25 +13,23 @@
...
-//! [grpc-includes]
-#include "google/cloud/storage/grpc_plugin.h"
-//! [grpc-includes]
 #include "google/cloud/internal/getenv.h"
+//! [grpc-includes] [START storage_grpc_quickstart]
+#include "google/cloud/storage/grpc_plugin.h"
+#include "google/cloud/storage/options.h"
+#include "google/cloud/common_options.h"
+#include "google/cloud/options.h"
+
```
Comment (devbww) [inline google/cloud/storage/examples/storage_grpc_samples.cc:22]: Remove extra blank line?  Or is it important in the final rendering?

**Review by devbww — APPROVED** [review]

File: google/cloud/storage/examples/storage_grpc_samples.cc (hunk 4 above)
Comment (coryan) [reply google/cloud/storage/examples/storage_grpc_samples.cc:24]: It is important in the final rendering. Added a comment to remind us in the future.

File: google/cloud/storage/examples/storage_grpc_samples.cc (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/examples/storage_grpc_samples.cc]: Fixed.

File: google/cloud/storage/examples/storage_grpc_samples.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/examples/storage_grpc_samples.cc]: Fixed.

File: google/cloud/storage/examples/storage_grpc_samples.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/examples/storage_grpc_samples.cc]: Fixed.

**Pull request merged**